src/test/java/com/greenfox/lvlup/controller/BadgeControllerTest.java

```diff
@@ -22,16 +34,37 @@
   @Autowired
   public MockMvc mvc;
 
-  private UserBadgeSetDTO testBadgeSet = new UserBadgeSetDTO();
-  private String tokenString = "testToken";
+  @InjectMocks
+  private BadgeController badgeController;
+
+  @MockBean
+  UserService userService;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+  }
 
   @Test
   public void showBadgesTestWithCorrectHeader() throws Exception {
-    mvc.perform(get("/badges")
-        .header("userTokenAuth", tokenString)
-        .contentType(MediaType.APPLICATION_JSON))
-        .andExpect(status().isOk())
-        .andExpect(content().string(stringify(testBadgeSet)));
+
+    UserBadgeDTO userBadgeDTO1 = new UserBadgeDTO("Process improver", 1);
+    UserBadgeDTO userBadgeDTO2 = new UserBadgeDTO("English speaker", 2);
+    List<UserBadgeDTO> userBadgeDTOList = new ArrayList<>();
+    userBadgeDTOList.add(userBadgeDTO1);
+    userBadgeDTOList.add(userBadgeDTO2);
+
+    List<BadgeLevel> badgeLevelsMock = new ArrayList<>();
+    badgeLevelsMock.add(0, new BadgeLevel(1));
+    badgeLevelsMock.add(1, new BadgeLevel(2));
+
+    when(userService.getBadges(anyLong())).thenReturn(badgeLevelsMock);
+    when(userService.convertBadgeToDTO(badgeLevelsMock)).thenReturn(userBadgeDTOList);
+
+    ResponseEntity testResult = badgeController.getUserBadges(1L, "testToken");
+
+    assertNotNull(testResult);
+    assertEquals(userBadgeDTOList, testResult.getBody());
```

Comment:
Like this I can almost aprove it!
Why almost? -> you mocked both the service methods which were used in that controller method, but:
 - `userService.convertBadgeToDTO()` will convert the response of the `userService.getBadges()` method, so it is not necessary to return a real object with the first method call, you could just return anything with it, because you already set up the return value of the `userService.convertBadgeToDTO()`
 - What I'm trying to say is, that creating a `badgeLevelsMock`, if you already set up the `userService.convertBadgeToDTO(badgeLevelsMock)` return value, is unnecessary, because of this, you could return anything at the `userService.getBadges(anyLong())` method call